tests/test_ble_client_edge_cases.py

```
@@ -15,207 +20,217 @@
+    with pytest.raises(BLEClient.BLEError, match="Cannot pair: BLE client not initialized"):
+        ble_client.pair()
+
+    with pytest.raises(
+        BLEClient.BLEError, match="Cannot get services: BLE client not initialized"
+    ):
+        ble_client._get_services()
+
+    with pytest.raises(
+        BLEClient.BLEError, match="Cannot start notify: BLE client not initialized"
+    ):
+        ble_client.start_notify("uuid", lambda *args: None)
```

Ruff (0.15.2)

[warning] Unused lambda argument: `args`

(ARG005)